components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -74,6 +78,38 @@ brave_wallet::SwapController* BraveWalletService::swap_controller() const {
   return swap_controller_.get();
 }
 
+std::vector<std::string> BraveWalletService::WalletAccountNames() const {
+  std::vector<std::string> account_names;
+  for (const auto& account_name_value :
+       prefs_->Get(kBraveWalletAccountNames)->GetList()) {
+    const std::string* account_name = account_name_value.GetIfString();
+    DCHECK(account_name) << "account name type should be string";
+    account_names.push_back(*account_name);
+  }
+  return account_names;
+}
+
+// TODO(Douglas): This need to be seperated into 2 separate methods
+// 1. SetInitialAccountName to set the initial account name/names
+// 2. AddNewAccountName to use ListPrefUpdate to add a new account name to the
+// bottom of list
+void BraveWalletService::UpdateAccountNames(
+    const std::vector<std::string>& account_names) {
+  std::vector<base::Value> account_names_list;
+  ListPrefUpdate update(prefs_, kBraveWalletAccountNames);
+  if (prefs_->HasPrefPath(kBraveWalletAccountNames)) {
+    update->ClearList();
+  }
+  for (const std::string& name : account_names) {
+    if (prefs_->HasPrefPath(kBraveWalletAccountNames)) {
+      update->Append(base::Value(name));
+    } else {
+      account_names_list.push_back(base::Value(name));
+      prefs_->Set(kBraveWalletAccountNames, base::Value(account_names_list));
+    }
+  }
```

Comment:
If we are going to do it as follow-up, please just make this function as simple
`prefs_->Set(kBraveWalletAccountNames, base::Value(account_names_list));` because the `Clear` and then `Append` to the empty list is no different than just `Set`

Reply:
Done sir! 😊